pkg/config/analysis/incluster/controller.go

```diff
@@ -0,0 +1,114 @@
+/*
+ Copyright Istio Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package incluster
+
+import (
+	"fmt"
+	"time"
+
+	v1alpha12 "istio.io/api/analysis/v1alpha1"
+	"istio.io/api/meta/v1alpha1"
+	"istio.io/istio/pilot/pkg/config/kube/arbitraryclient"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/status"
+	"istio.io/istio/pkg/config/analysis/analyzers"
+	"istio.io/istio/pkg/config/analysis/diag"
+	"istio.io/istio/pkg/config/analysis/local"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+	"istio.io/istio/pkg/kube"
+	"istio.io/pkg/log"
+)
+
+// Controller manages repeatedly running analyzers in istiod, and reporting results
+// via istio status fields.
+type Controller struct {
+	analyzer  *local.IstiodAnalyzer
+	statusctl *status.Controller
+}
+
+func NewController(stop <-chan struct{}, rwConfigStore, configController model.ConfigStoreCache,
+	kubeClient kube.Client, namespace string, statusManager *status.Manager) (*Controller, error) {
+	ia := local.NewIstiodAnalyzer(schema.MustBuildMetadata(configController.Schemas()), analyzers.AllCombined(),
+		"", resource.Namespace(namespace), func(name collection.Name) {}, true)
+	ia.AddSource(rwConfigStore)
+	ctx := status.NewIstioContext(stop)
+	store, err := arbitraryclient.NewForSchemas(ctx, kubeClient, "default",
+		"cluster.local", collections.All.Remove(rwConfigStore.Schemas().All()...))
```

Comment:
I'm not clear on where "cluster.local" is meant to come from... any ideas?

Reply:
`crdclient.New(s.kubeClient, args.Revision, args.RegistryOptions.KubeOptions.DomainSuffix)` is how the current client does it. How easy it will be to get access to those variables in this context is another question...

Reply:
ok, I have added domainSuffix (will push in a minute).  I don't think there is a way for the user to specify a different default ns though...